rundbcmd.py

```diff
@@ -0,0 +1,440 @@
+        refname: "{{ refname_str }} (4)"
+        ignore: False (5)
+        debugging: False
+      whith_items:
+        - "select systdate from dual"
+        - "select user from dual"
+      when: master_node|bool
+
+      (1) user_id / user_password - sys or system
+
+      (2) host - including instance number. ( i.e. tlrac1, tlrac2.ccci.org or ploradr.dr.cru.org etc. )
+
+      (3) expect_results - REQUIRED. If the cmd should produce an output: True, else False
+
+      (4) refname - name used in Ansible to reference these facts ( i.e. cmdfacts )
+
+      (5) ignore - (connection errors) is optional. If you know the
+          database may be down set ignore: True. If connection to the
+          database fails the module will not throw a fatal error
+          and continue. If set to False this module will fail if an error occurs.
+
+'''
+
+# Debugging will go to: cru-ansible-oracle/bin/.utils/debug.log
+msg = ""
+debugme = True
+default_refname = "cmdfacts"
+affirm = ['True','TRUE', True, 'true', 'T', 't', 'Yes', 'YES', 'yes', 'y', 'Y']
+db_home_name = "dbhome_1"
+debug_log = "" # os.path.expanduser("~/.debug.log")
+utils_settings_file = os.path.expanduser("~/.utils")
+cru_domain = ".ccci.org"
+dr_domain = ".dr.cru.org"
+non_rac_dbs = [ "dw", "dr" ]
+cur = None
+nasty_list = [
+            "truncate",
+            "drop"
+            "drop database"
+            ]
+
+def set_debug_log():
+    """
+    Set the debug_log value to write debugging messages
+    Debugging will go to: cru-ansible-oracle/bin/.utils/debug.log
+    """
+    global utils_settings_file
+    global debug_log
+    global debugme
+
+    if not debugme or debug_log:
+        return()
+
+    try:
+        with open(utils_settings_file, 'r') as f1:
+            line = f1.readline()
+            while line:
+                if 'ans_dir' in line:
+                    tmp = line.strip().split("=")[1]
+                    debug_log = tmp + "/bin/.utils/debug.log"
+                    return()
+
+                line = f1.readline()
+    except:
+        print("ans_dir not set in ~/.utils unable to write debug info to file")
+        pass
+
+
+def add_to_msg(a_msg):
+    """Add the arguement to the msg to be passed out"""
+    global msg
+
+    if msg:
+        msg = msg + " " + a_msg
+    else:
+        msg = a_msg
+
+
+def debugg(db_msg):
+    """if debugging is on add this to msg"""
+    global debug_log
+    global debugme
+    global affirm
+    print("debugging log: {}".format(debug_log or "Empty!"))
+    if debugme not in affirm:
+        return()
+
+    add_to_msg(db_msg)
+
+    if not debug_log:
+        set_debug_log()
+        if not debug_log:
+            add_to_msg("Error setting debug log. No debugging will be available.")
+        return()
+
+    try:
+        with open(debug_log, 'a') as f:
+            f.write(db_msg + "\n")
+    except:
+        pass
+
+    return()
+
+
+def convert_size(arg_size_bytes, vunit):
```

Comment:
Is this function needed? Couldn't find any references to it.

Reply:
Not for the purpose I created it for, to drop restore points, but its very handy and thought if we query sizes in the future it would be easy to implement if it already exists. I can see future uses.